tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -2627,7 +2627,14 @@ Status StorageManager::load_array_metadata(
   stats_->add_counter("read_array_meta_size", meta_size);
 
   // Deserialize metadata buffers
-  metadata->deserialize(metadata_buffs);
+  auto&& [st_metadata, deserialized_metadata]{
+      Metadata::deserialize(metadata_buffs)};
+
+  if (!st_metadata.ok()) {
+    return Status::Ok();
+  }
+
+  *metadata = *(deserialized_metadata.value());
```

Comment:
This assignment statement is necessary given the function signature, but indicates a non-C.41 practice. Tracking this through, the real problem is that array metadata is declared as `Metadata` rather than as `optional<Metadata>`, so it has to be default-constructed. (This might have been avoided by allocating and using a smart pointer in the past, but `optional` is a better solution for this member variable.)

This is just a comment, not a change request. Making this change will be rather extensive and deserves its own PR.

Reply:
Yes, this is the place where copy assignment is needed